README.md

```diff
@@ -1,24 +1,157 @@
+
+
+In some cases just providing a DriverDefinition YAML is not sufficient. 
+ 1) The default storage class created may not be enough. The default storage class is simply a StorageClass with provisioner field set to the plugin’s name. You can use your own custom StorageClass by making a StorageClass yaml file and passing the name of that file in the StorageClass field of the DriverDefinition YAML file. csi-certify will then use this YAML to create the StorageClass that is required to test dynamic provisioning on your plugin.
+ 
+ 2) If simply deploying your plugin through YAML files is not enough, users would need to write their own [TestDriver](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/testdriver.go#L31), similar to the [HostPath TestDriver](https://github.com/wongma7/csi-certify/blob/refactor/pkg/certify/driver/hostpath_driver.go). This testdriver can be placed in the [pkg/certify/driver](https://github.com/wongma7/csi-certify/tree/master/pkg/certify/driver) directory. (WIP, need to add a template/skeleton file so this is easier to do).
+ 
+### How to run the e2e tests
+ 
+To run e2e tests using a DriverDefintion YAML: 
+```
+go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=/var/run/kubernetes/admin.kubeconfig --driverdef=<Path To Driver Info YAML> -timeout=0
+``` 
+
+To run e2e tests using the TestDriver that you wrote: 
+```
+go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=/var/run/kubernetes/admin.kubeconfig --testdriver=<Name of TestDriver that you want to run> -timeout=0
+``` 
+
+Since we have both the DriverDefinition YAML and a TestDriver written for the HostPath plugin, we can run it using either way. The command to run e2e tests on the HostPath CSI plugin by passing a DriverDefinition YAML file would be: 
+
+```
+kubectl create -f pkg/certify/driver/manifests #To first Install the hostpath driver on your local cluster
+
+go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=/var/run/kubernetes/admin.kubeconfig --driverdef=../../pkg/certify/external/driver-def.yaml -timeout=0
+```
+
+To run e2e tests on the HostPath CSI plugin using the implemented TestDriver: 
+```
+go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=/var/run/kubernetes/admin.kubeconfig --testdriver=hostpath -timeout=0
+```
+
+## NFS TestDriver Example
+
+An [NFS TestDriver](https://github.com/wongma7/csi-certify/blob/refactor/pkg/certify/driver/nfs_driver.go) was implemented to run e2e tests on the [NFS CSI Plugin](https://github.com/kubernetes-csi/csi-driver-nfs)
+
+### Notes
+
+For any testdriver the required methods that need to be implemented are:
+ - `GetDriverInfo() *testsuites.DriverInfo`
+ - `SkipUnsupportedTest(pattern testpatterns.TestPattern)`
+ - `GetDynamicProvisionStorageClass(config *testsuites.PerTestConfig, fsType string) *storagev1.StorageClass`
```

Comment:
GetDynamicProvisionStorageClass and GetClaimSize are not really needed, only the other 3 are, the DynamicPVTestDriver interface is optional.

Should link to https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/testdriver.go somewhere. The code is documented but it should be briefly explained. Anyone writing a TestDriver should be aware of the interfaces they can implement. e.g. not implementing  DynamicPVTestDriver means your driver is not capable of dynamic provisioning so of course the provisioning suite will not run for it.

Reply:
Fixed, and added some clarification